v0/destinations/marketo/networkResponseHandler.js

```diff
@@ -0,0 +1,251 @@
+      throw new ErrorBuilder()
+        .setStatus(500)
+        .setMessage(
+          `Request Failed for Marketo, ${errors[0].message} (Retryable).${sourceMessage}`
+        )
+        .setDestinationResponse({ ...trimmedResponse, success: false })
+        .setStatTags({
+          destination: DESTINATION,
+          stage,
+          scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+          meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.META.RETRYABLE
+        })
+        .build();
+    }
+    // http success but data not present
+    throw new ErrorBuilder()
+      .setStatus(500)
+      .setMessage(`Request Failed for Marketo (Retryable).${sourceMessage}`)
+      .setDestinationResponse({
+        ...trimmedResponse,
+        success: false
+      })
+      .setStatTags({
+        destination: DESTINATION,
+        stage,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.META.RETRYABLE
+      })
+      .build();
+  }
+  // http failure cases
+  const { response } = clientResponse.response;
+  if (!response && clientResponse.response && clientResponse.response.code) {
+    const nodeSysErr = nodeSysErrorToStatus(clientResponse.response.code);
+    throw new ErrorBuilder()
+      .setStatus(nodeSysErr.status || 500)
+      .setMessage(`Error occured ${sourceMessage} Error: ${nodeSysErr.message}`)
+      .setStatTags({
+        destination: DESTINATION,
+        stage,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: getDynamicMeta(nodeSysErr.status || 500)
+      })
+      .build();
+  } else {
+    const temp = trimResponse(clientResponse.response);
+    throw new ErrorBuilder()
+      .setStatus(temp.status || 500)
+      .setMessage(`Error occured ${sourceMessage} Error: ${temp.statusText}`)
+      .setDestinationResponse({ ...temp, success: false })
+      .setStatTags({
+        destination: DESTINATION,
+        stage,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: getDynamicMeta(temp.status || 500)
+      })
+      .build();
+  }
+};
+
+const sendGetRequest = async (url, options) => {
+  let clientResponse;
+  try {
+    const response = await httpGET(url, options);
+    clientResponse = { success: true, response };
+  } catch (err) {
+    clientResponse = { success: false, response: err };
+  }
+  return clientResponse;
+};
+
+const sendPostRequest = async (url, data, options) => {
+  let clientResponse;
+  try {
+    const response = await httpPOST(url, data, options);
+    clientResponse = { success: true, response };
+  } catch (err) {
+    clientResponse = { success: false, response: err };
+  }
+  return clientResponse;
+};
+
+const responseTransform = destResponse => {
+  let respBody;
+  try {
+    respBody = JSON.parse(destResponse.Body);
+  } catch (err) {
+    respBody = isEmpty(!destResponse.Body) ? destResponse.Body : null;
+  }
+  if (respBody && !respBody.success) {
```

Comment:
How do we know if the response is a success or failure here ? Where is this success property being populated ?

Reply:
respBody is the destination response for Marketo which will contain the success key in it specifying if the request is successful or failure. As it returns 200 for every request we need to check the success/failure scenario using this key

Reply:
Got it